idb/migration/migration_test.go

```diff
@@ -329,3 +330,105 @@ func TestSuccessfulMigration(t *testing.T) {
 		})
 	}
 }
+
+// TestAvailabilityChannelCloses tests that the migration object closes the availability
+// channel when blocking migrations finish. If it is not the case, this test will deadlock
+// (and timeout).
+func TestAvailabilityChannelCloses(t *testing.T) {
+	// Migration 2 reads on this channel.
+	migrationTwoChannel := make(chan struct{})
+
+	tasks := []Task{
+		{
+			MigrationID: 1,
+			Handler: func() error {
+				return nil
+			},
+			DBUnavailable: true,
+		},
+		{
+			MigrationID: 2,
+			Handler: func() error {
+				<-migrationTwoChannel
+				return nil
+			},
+		},
+	}
+
+	m, err := MakeMigration(tasks, nil)
+	require.NoError(t, err)
+
+	availableCh := m.RunMigrations()
+	_, ok := <-availableCh
+	assert.False(t, ok)
+
+	// This will block until migration 1 reads from the channel.
+	migrationTwoChannel <- struct{}{}
+}
+
+// TestAvailabilityChannelClosesNoMigrations tests that the migration object closes
+// the availability channel when last migration, which is blocking, finishes.
+// If it is not the case, this test will deadlock (and timeout).
+func TestAvailabilityChannelClosesNoMigrations(t *testing.T) {
+	tasks := []Task{
+		{
+			MigrationID: 1,
+			Handler: func() error {
+				return nil
+			},
+			DBUnavailable: true,
+		},
+	}
+
+	m, err := MakeMigration(tasks, nil)
+	require.NoError(t, err)
+
+	availableCh := m.RunMigrations()
+	_, ok := <-availableCh
+	assert.False(t, ok)
```

Comment:
Don't you want to verify that it completed?
```suggestion
	select {
	case <-availableCh:
		
	case <- time.After(2 * time.Second):
		assert.Fail("The channel have been closed by now.")
	}
```

Reply:
The test runner would timeout. Are you proposing to explicitly set a smaller timeout?

Reply:
Yes, but also I'm not exactly sure what `_, ok := <-availableCh` is supposed to do.

Reply:
I added timeouts.